sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveInlineTables.scala

```diff
@@ -33,12 +34,14 @@ import org.apache.spark.sql.types.{StructField, StructType}
  */
 object ResolveInlineTables extends Rule[LogicalPlan]
   with CastSupport with AliasHelper with EvalHelper {
-  override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsWithPruning(
-    AlwaysProcess.fn, ruleId) {
-    case table: UnresolvedInlineTable if table.expressionsResolved =>
-      validateInputDimension(table)
-      validateInputEvaluable(table)
-      convert(table)
+  override def apply(plan: LogicalPlan): LogicalPlan = {
+    plan.resolveOperatorsDown(p => ComputeCurrentTime(p)).resolveOperatorsWithPruning(
```

Comment:
`ComputeCurrentTime` transforms the entire plan tree, so the code should be `ComputeCurrentTime(plan).resolveOperatorsWithPruning ...`

Reply:
I did try that originally but I got:
`      throw QueryExecutionErrors.methodCalledInAnalyzerNotAllowedError()` 

I ended up with resolveOperatorsDown due to this comment:
```scala
  /**
   * In analyzer, use [[resolveOperatorsDown()]] instead. If this is used in the analyzer,
   * an exception will be thrown in test mode. It is however OK to call this function within
   * the scope of a [[resolveOperatorsDown()]] call.
   * @see [[org.apache.spark.sql.catalyst.trees.TreeNode.transformDownWithPruning()]].
   */
  override def transformDownWithPruning(cond: TreePatternBits => Boolean,

```

Reply:
We can wrap the code with `AnalysisHelper.allowInvokingTransformsInAnalyzer` to bypass this limitation.

Reply:
I think that my current approach is not valid. i.e. if we do ComputeCurrentTime here we will break following assumption:

// Technically some of the rules in Finish Analysis are not optimizer rules and belong more
// in the analyzer, because they are needed for correctness (e.g. ComputeCurrentTime).
// However, because we also use the analyzer to canonicalized queries (for view definition),
// we do not eliminate subqueries or compute current time in the analyzer.

So, if we had inline table in view definition this wouldn't work (although this is currently incorrect as well). 

On the other hand, if we were to replace current time only for inline table values then we might end up with different results for CURRENT_DATE/TIMESTAMP in other parts of the plan so that doesn't work either.

I will see if I can delay inline table evaluation, or at least delay it up for expressions for which we can guarantee that they will be replaced by a literal. This PR is on hold until then.

Reply:
Oh good point. We should have a logical plan for inline tables to keep the expressions, and then replace it with `LocalRelation` after we run `ComputeCurrentTime`

Reply:
I played a bit more and came up with solution that is in current PR (which I don't like that much to be honest, since it adds another rule in post analyzer phase).

Here is the rough logic:
1) In `ResolveInlineTable` we make sure that shape is correct and that there is proper casting for all the objects. Here we also decide whether all expressions can be reduced to `LocalRelation` or we need to postpone that step. Option here is to always postpone - I originally went with that approach but that changed bunch of GOLDEN files since we now have TempResolvedInlineTable after every analysis run. But still, it is cleaner to always postpone - looking for feedback here.
2) In EvalInlineTable, after ReplaceCurrentTime/Like we do final evaluation if needed.

By doing this we can solve a couple of additional "bugs" - e.g. when creating view as CREATE VIEW V AS (SELECT * FROM VALUES (NOW())) previous behaviour was to embed time of view creation in query, while proper thing should be to calculate time on every invocation of the view (I verified that PG does this).

An option is to also do everything in `EvalInlineTable` rule. The problem here is that I think we will want at least to check whether query is correct in analysis phase (i.e. check casting and shape) and then do final eval() in finishAnalysis phase, which is part of optimizer.

In short, we should have short discussion about this. @cloud-fan , @MaxGekk, @beliefer, @dtenedor and @srielau - please provide feedback.